posthog-ai/src/langchain/callbacks.ts

```diff
@@ -0,0 +1,578 @@
+        generation.model = metadata.ls_model_name
+      }
+      if (metadata.ls_provider) {
+        generation.provider = metadata.ls_provider
+      }
+    }
+    if (serialized && 'kwargs' in serialized && serialized.kwargs.openai_api_base) {
+      generation.baseUrl = serialized.kwargs.openai_api_base
+    }
+    this.runs[runId] = generation
+  }
+
+  private _popRunMetadata(runId: string): RunMetadata | undefined {
+    const endTime = Date.now()
+    const run = this.runs[runId]
+    if (!run) {
+      console.warn(`No run metadata found for run ${runId}`)
+      return undefined
+    }
+    run.endTime = endTime
+    delete this.runs[runId]
+    return run
+  }
+
+  private _getTraceId(runId: string): string {
+    return this.traceId ? String(this.traceId) : this._findRootRun(runId)
+  }
+
+  private _getParentRunId(traceId: string, runId: string, parentRunId?: string): string | undefined {
+    // Replace the parent-run if not found in our stored parent tree.
+    if (parentRunId && !this.parentTree[parentRunId]) {
+      return traceId
+    }
+    return parentRunId
+  }
+
+  private _popRunAndCaptureTraceOrSpan(
+    runId: string,
+    parentRunId: string | undefined,
+    outputs: ChainValues | DocumentInterface[] | AgentFinish | Error | any
+  ): void {
+    const traceId = this._getTraceId(runId)
+    this._popParentOfRun(runId)
+    const run = this._popRunMetadata(runId)
+    if (!run) return
+    if ('modelParams' in run) {
+      console.warn(`Run ${runId} is a generation, but attempted to be captured as a trace/span.`)
+      return
+    }
+    const actualParentRunId = this._getParentRunId(traceId, runId, parentRunId)
+    this._captureTraceOrSpan(traceId, runId, run as SpanMetadata, outputs, actualParentRunId)
+  }
+
+  private _captureTraceOrSpan(
+    traceId: string,
+    runId: string,
+    run: SpanMetadata,
+    outputs: ChainValues | DocumentInterface[] | AgentFinish | Error | any,
+    parentRunId?: string
+  ): void {
+    const eventName = parentRunId ? '$ai_span' : '$ai_trace'
+    const latency = run.endTime ? (run.endTime - run.startTime) / 1000 : 0
+    const eventProperties: Record<string, any> = {
+      $ai_trace_id: traceId,
+      $ai_input_state: withPrivacyMode(this.client, this.privacyMode, run.input),
+      $ai_latency: latency,
+      $ai_span_name: run.name,
+      $ai_span_id: runId,
+    }
+    if (parentRunId) {
+      eventProperties['$ai_parent_id'] = parentRunId
+    }
+
+    Object.assign(eventProperties, this.properties)
+    if (!this.distinctId) {
+      eventProperties['$process_person_profile'] = false
+    }
+    if (outputs instanceof Error) {
+      eventProperties['$ai_error'] = outputs.toString()
+      eventProperties['$ai_is_error'] = true
+    } else if (outputs !== undefined) {
+      eventProperties['$ai_output_state'] = withPrivacyMode(this.client, this.privacyMode, outputs)
+    }
+    this.client.capture({
+      distinctId: (this.distinctId as string) || runId,
```

Comment:
```suggestion
      distinctId:  this.distinctId ? String(this.distinctId) : runId,
```

Two things:
- `distinctId` can be theoretically 0, which we won't capture.
- `distinctId` by the type is a string. A number might break things.